lib/src/utils/utils.dart

```diff
@@ -137,3 +137,9 @@ bool isExpectConversion(int tag) {
 
   return false;
 }
+
+const bool kIsWeb = identical(0, 0.0);
+
+bool isWebDouble(Object a) {
+  return a is double && a.toInt() == a;

```

Comment:
This would return `true` any floating point number intentionally used to store integers, i.e. `var d = 1.0`.